modules/doc/content/getting_started/installation/inl_hpc_install_moose.md

```diff
@@ -1,71 +1,109 @@
+development environment to use. For the purposes of this documentation, we will use the version
+[!versioner!version package=moose-dev], although yours could be different!
 
-## Build and Test MOOSE
+To enter the containerized shell environment to proceed with building, run the following
+(note that here we are requesting the specific version of the environment that we need):
 
-!style! halign=left
-!template load file=installation/build_moose.md.template PATH=~/cluster_name/projects
-!style-end!
+!versioner! code
+module load use.moose moose-dev-openmpi/__VERSIONER_VERSION_MOOSE_DEV__
+moose-dev-shell
+!versioner-end!
 
-!template load file=installation/test_moose.md.template PATH=~/cluster_name/projects
+You are now within the versioned, containerized environment. The standard
+MOOSE dependencies (PETSc, libMesh, WASP, etc) are built and there is not a need to run
+any of the update-and-rebuild scripts within MOOSE. You +must+ repeat the commands
+above whenever you build any code.
 
-Now that you have a working MOOSE, proceed to 'New Users' to begin your tour of MOOSE!
+At this point, we have assumed that you already have the source code for your application
+available and are in the directory that contains that application. Build your application
+and test it as normal:
 
-!content pagination use_title=True
-                    previous=installation/inl_hpc_index.md
-                    next=getting_started/inl_hpc_new_users.md
\ No newline at end of file
+```
+make -j 6
+./run_tests -j 6
+```
+
+## Running
+
+How your application is ran is dependent on whether or not you are running across multiple
+hosts. In general, you are running across multiple hosts when you are running within a job
+on HPC. We have two different methods of execution that we will reference later:
+
+- Shell: Enters an interactive shell within the container where you can run multiple commands within the environment, can only be used on a single host
+- Execute: Runs a single command within the container, required for running on multiple hosts
+
+When running on a single host (i.e., not in a HPC job), enter the container in a shell as
+done above and execute your application as normal:
+
+!versioner! code
+module load use.moose moose-dev-openmpi/__VERSIONER_VERSION_MOOSE_DEV__
+moose-dev-shell
+mpiexec -n 2 /path/to/your/application-opt -i ...
+!versioner-end!
+
+When you are running on multiple hosts (i.e., in a HPC job) you must used the "execute"
+method to run your application as described above. This is done by appending the command
+`moose-dev-exec` before the path to the binary of your application. That is, within a HPC
+job submission script or within a HPC job interactive session, you will run your application
+with:
+
+!versioner! code
+module load use.moose moose-dev-openmpi/__VERSIONER_VERSION_MOOSE_DEV__
+mpiexec -n 2 moose-dev-exec /path/to/your/application-opt -i ...
+!versioner-end!
+
+Note that the "execute" method will also work on a single host.
```

Comment:
Maybe want to add comments on why OpenMPI is chosen, how if users want to change MPI package, how if users want to do profiling, etc.